debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java

```diff
@@ -414,6 +414,16 @@ public static SchemaNameAdjustmentMode parse(String value) {
                     "This doesn't affect the snapshot events' values, but the schema of snapshot events may have outdated defaults.")
             .withDefault(Boolean.FALSE);
 
+    public static final Field INCREMENTAL_SNAPSHOT_CHUNK_KEY_COLUMNS = Field.create("incremental.snapshot.chunk.key.columns")
+            .withDisplayName("List of columns to be used by query during incremental snapshot.")
+            .withType(Type.STRING)
+            .withWidth(Width.LONG)
+            .withImportance(Importance.LOW)
+            .withValidation(Field::isListOfRegex)
+            .withDescription("This settings can be set to specify a list of columns for specific tables that would be used " +
+                    "to perform filtering when running an incremental snapshot.")
+            .withDefault(Boolean.FALSE);
+
```

Comment:
This is probably the one aspect I'm a bit torn on.  I'm wondering if it would be better if this was something specified as a part of the incremental snapshot signal rather than a connector configuration proper.  I can possibly see an argument either way, so perhaps @jpechane can be our deciding vote :).

Reply:
Went with this way since it's a configuration that is not going to change that often, high probability that if it's being set, it won't change since users would like to have the queries every time they run an incremental snapshot. 
Anyway, open to suggestions.

Reply:
@PlugaruT Please make this part of the signal payload. @Naros is right that this belongs to it.

Reply:
Gotcha, I'll see how I should do that. Thanks for the input.

Reply:
@jpechane Looked at the code a bit, by adding this new config to the signal, it will mean that we would change the structure of the signal. Will that be ok?

Reply:
@PlugaruT If this will be done in backward compatible way (like new optional field) then there is no issue in that.

Reply:
@jpechane I will bother you with a few questions of how to implement this change as the signalling approach is a bit more tricky. 
From my understanding, this is what happens when an incremental snapshot get's triggered:
 1. Signal get's processed by the EventDispatcher that contains needed information for the snapshot, including the `data-collections` which represents the list of tables to be snapshoted.
 2. A `IncrementalSnapshotContext` object get's instantiated that holds the information about the snapshot, like max keys and other configurations for the snapshot. From my understanding, this context get's also stored into the signalling table because this context is re-used in case the connector is re-started while the snapshot is in progress. If not the entire context, then again, information about the chunk sizes and the limits.
 
From my understanding, this new configuration should also be stored inside the signalling table because it should be persistent for the entire duration of the snapshot as the signal that triggers the snapshot is only processed once. 

Did I get it right? Am I looking into the right direction stil?

I will do my best to implement this, but it will take a while until I get my head around of how the signalling architecture is implemented 😅 
Thanks

Reply:
@PlugaruT You've it almost right. The context *is* persisted but not in signalling table but in Kafka Connect offsets. See e.g. `io.debezium.connector.postgresql.PostgresOffsetContext.getOffset()`. That provides a an offset associated with each message sent to Kafka Connect which will then persist it.